### PR TITLE
Add unit tests and update methods for ticket management with Mockito

### PR DESCRIPTION
**Pull request by Arthur2224**
- Added tests for ticket update functionality, covering valid, invalid, and non-existing ticket scenarios.
- Updated the `updateTicket()` method to handle null ticket input and null ID validation.
- Used Mockito to mock `ticketDAO` behavior and ensure correct interactions.
- Verified that the update method is not called for non-existing tickets by mocking the `findById` method.
**_All tests have passed_**

### CODE REVIEW

File: src/test/java/com/andersen/TicketService/TicketServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    // Setup for each test. Initializes mock objects and the service under test.
+    @BeforeEach
+    void setup() {
+        ticketDAO = mock(TicketDAO.class);
+        ticketService = new TicketService(ticketDAO);
+    }
+
+    // Test for adding a ticket. Includes positive, negative, and corner cases.
+    @Test
+    void testAddTicket() {
```
Comment (Darya-Drutsko) [inline src/test/java/com/andersen/TicketService/TicketServiceTest.java]: for each case create please separate test, also don't repeate 'test' in method name inside Test class

File: src/test/java/com/andersen/TicketService/TicketServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        // Positive case: Valid ticket is added successfully
+        Ticket ticket = new Ticket();
+        ticket.setId(1L);
+        ticketService.addTicket(ticket);
```
Comment (Darya-Drutsko) [inline src/test/java/com/andersen/TicketService/TicketServiceTest.java]: Also we could check return value for this case

**Review by Darya-Drutsko — COMMENTED** [review]
Left comments which are applicable to all tests)

**Review by Darya-Drutsko — APPROVED** [review]
Looks good, just overall comment -> usually we're not using comments for describing the test, method name should be enough. Also we could highlight negative test in name - e.g. addTicket_NullTicket -> fail_addNullTicket, and avoid duplication in method names :)
